src/software/jetson_nano/services/network/network.cpp

```diff
@@ -74,3 +77,66 @@ void NetworkService::primitiveSetCallback(TbotsProto::PrimitiveSet input)
                      << "%";
     }
 }
+
+void NetworkService::logNewPrimitiveSet(TbotsProto::PrimitiveSet input)
+{
+    /* TODO: THIS TRACKS THE RECEIVED PRIMITIVES
+     *  - make sure it is in the lock
+     *  - log the new PRIMITIVE SET PROTO within a deque if it is valid (see primitive_tracker.send())
+     *  - update the timestamp on each primitive set to the current epoch time
+     */
+
+    if (primitive_set_tuple_rtt.size() >= 50)
+    {
+        LOG(WARNING) << "Too many primitive sets logged for round-trip calculations, halting log process";
+        return;
+    }
```

Comment:
when/why would this happen?

Reply:
This is purely here to prevent the possibility of a memory leak since thunderloop will be receiving primitive sets at 60Hz, whereas the "cleanup" of the deque occurs at only 30Hz. In theory, an infinitely growing deque is not possible, but in the slight possibility that some error in thunderloop interferes with the creation of a new robot status, which could then break the cleanup process.

Reply:
Updated to scale with Thunderscope refresh rate